### DISCUSSION TITLE
Issue On. Converting HTML to PDF

### DISCUSSION THREAD

**Discussion by nidhintrell** [mpdf/mpdf#1947] [Q&A] [answered]
I'm trying to convert the html in [this link](https://jsfiddle.net/yj3td7wf/1/) to pdf using mpdf. but except for 1 class everything is fine. attaching the screenshot for desired output and current output

DESIRED OUTPUT
<img width="750" alt="Screenshot 2023-10-29 at 11 58 15 PM" src="https://github.com/mpdf/mpdf/assets/67859556/5243f130-7c92-4348-97e4-5afe6d7497f7">

Actual Output 
<img width="586" alt="Screenshot 2023-10-29 at 11 52 20 PM" src="https://github.com/mpdf/mpdf/assets/67859556/7742ce8b-c6fe-4e2c-b962-989fea6d2f58">

the 1st circle div is not getting aligned to centre horizontaly. 
Where ami i going wrong here. it looks fine in browser

**Comment by dgaidula** [comment, marked as answer]
this is easy mPDF doesn't support display:flex and never will. See this from the docs.

https://github.com/mpdf/mpdf#user-content-about-css-support-and-development-state

https://mpdf.github.io/css-stylesheets/supported-css.html

While mPDF supports quite a lot of CSS, if you're going to use in production you are to double-check all of your HTML/CSS and make sure that you are writing it to mPDF's spec of supported CSS and HTML.

I hope this helps and like it says, mPDF is quite old and any new project should just use newer more reliable methods.

**Reply by nidhinprathap** [reply]
@dgaidula any suggestions for any alternative html to pdf libraries ?
